enhancements/microshift/disabling-storage-components.md

```diff
@@ -0,0 +1,244 @@
+5. If all checks pass, LVMS and CSI are deployed.
+
+**_Installation without LVMS and Snapshotting_**
+
+1. User installs MicroShift.
+2. User specifies a MicroShift config sets fields to disable LVMS and CSI Snapshots.
+3. MicroShift starts and detects the provided config.
+4. LVMS and CSI snapshot components are not deployed.
+
+**_Post-Start Installation, LVMS only_**
+
+1. User has already installed and started MicroShift service.
+2. User later determines there is a requirement for persistent storage, but not snapshotting.
+3. User edits the MicroShift config, enabling LVMS only.
+4. User restarts MicroShift service.
+5. MicroShift starts and detects the provided config.
+6. MicroShift performs LVMS startup checks.
+7. If all checks pass, LVMS is deployed. Else if checks fail, an error is logged, LVMS deployment is skipped, 
+and startup continues.
+
+**_Complete Uninstallation, with Data removal_**
+
+> NOTE: Installation is not reversible. Deleting LVMS or the CSI snapshotter while there are still storage volumes risks
+orphaning volumes. It is the user's responsibility to manually destroy data volumes and/or snapshots before
+uninstalling.
+
+1. User has already deployed a cluster with LVMS and CSI Snapshotting installed.
+2. User has deployed cluster workloads with persistent storage.
+3. User decides that LVMS and snapshotting are no longer needed.
+4. User edits `/etc/microshift/config.yaml`, setting `.storage.driver: none`. 
+5. User takes actions to back up, wipe, or otherwise ensure data will not be irretrievable.
+6. User stops workloads with mounted storage volumes.
+7. User deletes VolumeSnapshots and waits for deletion of VolumeSnapshotContent objects to verify. The deletion process
+cannot happen after the CSI Snapshotter is deleted.
+8. User delete PersistentVolumeClaims and waits for deletion of PersistentVolumes. The deletion process
+cannot happen after LVMS is deleted.
+9. User deletes the relevant LVMS and Snapshotter cluster API resources:
+    ```shell
+    $ oc delete -n kube-system deployment.apps/csi-snapshot-controller deployment.apps/csi-snapshot-webhook
+    $ oc delete -n openshift-storage daemonset.apps/topolvm-node
+    $ oc delete -n openshift-storage deployment.apps/topolvm-controller
+    $ oc delete -n openshift-storage configmaps/lvmd
+    $ oc delete storageclasses.storage.k8s.io/topolvm-provisioner
+    ```
+10. Component deletion completes. On restart, MicroShift will not deploy LVMS or the snapshotter.
+
+### API Extensions
+
+- MicroShift config will be extended from the root with a field called `storage`, which will have 2 subfields.
+  - `.storage.driver`: **ENUM**, type is preferable to leave room for future supported drivers
+    - One of ["None|none", "lvms"]
+    - An empty value or null field defaults to deploying LVMS. This is because the driver is already. 
+  - `.storage.with-csi-components:` **Array**. Empty array or null value defaults to not deploying additional CSI 
```

Comment:
maybe we should set it as a dedicated key so that if additional settings for the controller come up, we can directly key them under there, e.g. .storage.snapshot.enabled?

Just a thought, leaving it up to you

Reply:
Just read https://github.com/openshift/enhancements/pull/1653#discussion_r1677941507 so same thought I guess